contracts/SpokePool.sol

```diff
@@ -268,6 +272,61 @@ abstract contract SpokePool is Testable, Lockable, MultiCaller {
         );
     }
 
+    function increaseRelayFee(
+        address depositor,
+        address recipient,
+        address destinationToken,
+        uint64 realizedLpFeePct,
+        uint64 relayerFeePct,
+        uint64 newRelayerFeePct,
+        uint64 depositId,
+        uint256 originChainId,
+        uint256 totalRelayAmount,
+        bytes32 depositorMessageHash,
+        bytes memory depositorSignature
+    ) public nonReentrant {
+        RelayData memory relayData = RelayData({
+            depositor: depositor,
+            recipient: recipient,
+            destinationToken: destinationToken,
+            realizedLpFeePct: realizedLpFeePct,
+            relayerFeePct: relayerFeePct,
+            depositId: depositId,
+            originChainId: originChainId,
+            relayAmount: totalRelayAmount
+        });
+        bytes32 relayHash = _getRelayHash(relayData);
+
+        // By requiring that the new fee is higher, we remove the possibility that the depositor signs a message and
+        // calls this method with a lower fee, dropping the relay fee at the last second unbeknownst to the relayer.
+        require(newRelayerFeePct >= relayerFeePct, "new fee cannot be lower");
+
+        // Check if relay is not yet fully filled.
+        // Note: we cannot easily check that this relay is actually pending, as the relay hash above can point to an
+        // uninitialized relay, and this function will not revert as long as the signed message and signature correspond
+        // to the depositor's address, and the updated relayer fee is contained in the message. This is not a problem
+        // because relayers who store relay data for deposits that don't exist will not get rewarded. They are instead
+        // just wasting gas to submit this transaction.
+        require(relayFills[relayHash] < totalRelayAmount, "filled relay");
+
+        // Depositor should have signed a hash of the relayer fee % to update to.
+        bytes32 expectedDepositorMessageHash = keccak256(abi.encode(newRelayerFeePct));
```

Comment:
I think we need more than this to avoid replay attacks. For instance, if I agree to a relayer fee of 23% on a small relay, then I send a large relay, someone could take my signed message from the small relay and replay it to get a large fee from me in the in the large relay.

A few additional fields I think we need: depositId, chainId. We might also want include a constant value in front, like "ACROSS-V2 FEE 1.0" or something that we could increment if we ever updated spoke pools and ended up resetting the depositId counter.